front/src/value_dialog.tsx

```diff
@@ -0,0 +1,397 @@
+            children_factory={props.children_factory}
+            title={props.title}
+            isValidValue={props.isValidValue}
+            context_children={
+                <>
+                    {contextError && (
+                        <Alert severity='error'>
+                            {contextError}
+                            {existingRuleInError !== null && (
+                                <Button
+                                    onClick={() => {
+                                        props.onClose(false);
+                                        props.onTriggerChange(existingRuleInError);
+                                    }}
+                                >
+                                    Edit rule
+                                </Button>
+                            )}
+                        </Alert>
+                    )}
+                    <div style={{ justifyContent: 'center', width: '100%' }}>
+                        <ContextSelect
+                            context_options={props.contextOptions!}
+                            filterChangeCallback={changeContext}
+                            initialValue={initial_context}
+                            isConcrete={true}
+                            stackProps={{ spacing: 1, alignItems: 'center' }}
+                        />
+                    </div>
+                </>
+            }
+            info_area={
+                <TextField onChange={(e) => props.onInfoChange(e.target.value)} sx={{ py: 1 }} label='Info' fullWidth />
+            }
+            contextError={contextError}
+        />
+    );
+}
+
+export function ValueEditDialogConstContext(props: ValueEditDialogConstContextProps) {
+    // all contexts that are "undefined" in the partial context should default to *
+    const context = new Map(
+        Array.from(props.contextFeatures).map((k) => {
+            let v = props.initialContext!.get(k);
+            if (v === undefined || v === '<none>') {
+                v = '*';
+            }
+            return [k, v];
+        }),
+    );
+
+    return (
+        <BaseValueEditDialog
+            open={props.open}
+            onClose={props.onClose}
+            initial_value={props.initial_value}
+            on_value_changed={props.on_value_changed}
+            on_value_validity_changed={props.on_validity_changed}
+            children_factory={props.children_factory}
+            title={props.title}
+            isValidValue={props.isValidValue}
+            context_children={
+                <Typography sx={{ ml: 3 }}>
+                    {Array.from(context.entries())
+                        .map(([k, v]) => `${k}: ${v}`)
+                        .join(', ')}
+                </Typography>
+            }
+            info_area={null}
+            contextError={''}
+        />
+    );
+}
+
+export function ValueEditDialogNoContext(props: ValueEditDialogNoContextProps) {
```

Comment:
find a way to combine this component and the one from the above. Instead of creating 2 different components you can control it with a boolean prop.